Source/SelfService/Web/microservice/deployMicroservice/deployMicroservice.tsx

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import React, { useState } from 'react';
+import { useHistory } from 'react-router-dom';
+
+import { useSnackbar } from 'notistack';
+
+import { Typography } from '@mui/material';
+import { RocketLaunch } from '@mui/icons-material';
+
+import { Guid } from '@dolittle/rudiments';
+import { Button, Form, LoadingSpinner } from '@dolittle/design-system';
+
+import { saveSimpleMicroservice } from '../../stores/microservice';
+
+import { MicroserviceSimple, MicroserviceFormParameters } from '../../api/index';
+import { getLatestRuntimeInfo, getRuntimes } from '../../api/api';
+import { HttpResponseApplication } from '../../api/application';
+
+import { HasM3ConnectorField } from '../components/form/hasM3ConnectorField';
+import { SetupFields } from './setupFields';
+import { ContainerImageFields } from './containerImageFields';
+import { PublicUrlFields } from './publicUrlFields';
+import { getRuntimeNumberFromString } from '../helpers';
+
+const styles = {
+    form: {
+        'mt': 4.5,
+        'ml': 3,
+        '& .MuiFormControl-root': {
+            my: 1
+        },
+        '.MuiFormControlLabel-root': {
+            ml: 0
+        }
+    },
+    formSections: {
+        'mb': 4,
+        'display': 'flex',
+        'flexDirection': 'column',
+        '&:last-child': {
+            mb: 0
+        }
+    }
+};
+
+type DeployMicroserviceProps = {
+    application: HttpResponseApplication;
+    environment: string;
+};
+
+export const DeployMicroservice = ({ application, environment }: DeployMicroserviceProps) => {
+    const { enqueueSnackbar } = useSnackbar();
+    const history = useHistory();
+
+    const [isLoading, setIsLoading] = useState(false);
+    const [showM3ConnectorInfo, setShowM3ConnectorInfo] = useState(false);
+
+    const environmentInfo = application.environments.find(env => env.name === environment)!;
```

Comment:
I don't think this is a performance issue, but this value could potentially be memorized using a `useMemo`-hook.